stl/inc/format

```diff
@@ -240,6 +244,69 @@ constexpr const _CharT* _Parse_precision(const _CharT* _Begin, const _CharT* _En
     return _Begin;
 }
 
+template <class _CharT, _Parse_spec_callbacks<_CharT> _Callbacks_type>
+constexpr const _CharT* _Parse_format_specs(const _CharT* _Begin, const _CharT* _End, _Callbacks_type&& _Callbacks) {
+    if (_Begin == _End || *_Begin == '}') {
+        return _Begin;
+    }
+
+    _Begin = _Parse_align(_Begin, _End, _Callbacks);
+    if (_Begin == _End) {
+        return _Begin;
+    }
+
+    switch (*_Begin) {
+    case '+':
+        _Callbacks._On_sign(_Sign::_Plus);
+        ++_Begin;
+        break;
+    case '-':
+        _Callbacks._On_sign(_Sign::_Minus);
+        ++_Begin;
+        break;
+    case ' ':
+        _Callbacks._On_sign(_Sign::_Space);
+        ++_Begin;
+        break;
+    }
+
+    if (_Begin == _End) {
+        return _Begin;
+    }
+
+    // Parse hash.
+    if (*_Begin == '#') {
+        _Callbacks._On_hash();
+        if (++_Begin == _End) {
+            return _Begin;
+        }
+    }
+
+    // Parse zero flag
+    if (*_Begin == '0') {
+        _Callbacks._On_zero();
+        if (++_Begin == _End) {
+            return _Begin;
+        }
+    }
+
+    _Begin = _Parse_width(_Begin, _End, _Callbacks);
+    if (_Begin == _End) {
+        return _Begin;
+    }
+
+    // Parse precision
+    if (*_Begin == '.') {
+        _Begin = _Parse_precision(_Begin, _End, _Callbacks);
+    }

```

Comment:
I think it would help readability if this was:
```c++
_Begin = _Parse_precision(_Begin, _End, _Callbacks);
if (_Begin == _End) {
    return _Begin;
}
```
just like 253-256 and 293-296, i.e., with the check for `.` inside `_Parse_precision`. Do you agree?

Reply:
not really, neither width nor alignment have any kind of "control character" marking to indicate that they are included, they are just there or not there. This aligns with what libfmt does as well.

Reply:
> This aligns with what libfmt does as well.

Being consistent with upstream's inconsistency works for me, if just barely ;)